lib/src/tar/export.rs

```diff
@@ -281,7 +281,7 @@ impl<'a, W: std::io::Write> OstreeTarWriter<'a, W> {
         h.set_uid(meta.attribute_uint32("unix::uid") as u64);
         h.set_gid(meta.attribute_uint32("unix::gid") as u64);
         let mode = meta.attribute_uint32("unix::mode");
-        h.set_mode(mode);
+        h.set_mode(mode & 0o7777);
```

Comment:
Let's use `& !libc::S_IFMT` for greppability?

Reply:
Ack, done!